table/merging_iterator.cc

```diff
@@ -1304,6 +1304,10 @@ void MergeIteratorBuilder::AddRangeTombstoneIterator(
       first_iter = nullptr;
     }
   }
+  if (to_add_memtable_range_tombstone_iter_) {
+    merge_iter->AddRangeTombstoneIterator(nullptr);
+    to_add_memtable_range_tombstone_iter_ = false;
+  }
```

Comment:
The assumptions on the caller are reasonable but adding up to be somewhat complicated IMO.

What if we add `bool ignore_range_deletions` to the `MergeIteratorBuilder` constructor, and then have a common `SwitchToMergingIterator()` function that does everything when needed - create the MergingIterator, add first_iter as first point key iterator, add nullptr as first range del iterator (when !ignore_range_deletions)?

Reply:
Perhaps a better design is having a single AddComponentIterators() function that takes both point and range iterator pointers. It could internally decide when to upgrade to MergingIterator (upon first call when range tombstone iter is non-null or second call otherwise). It could also have some logic to only call `AddRangeTombstoneIterator()` upon the first non-nullptr range iterator provided - first calling `AddRangeTombstoneIterator(nullptr)` for all the previous components for which we have not yet added a range tombstone iterator.

Reply:
Thanks for the review. I agree with the complexity involved in `MergeIteratorBuilder`, and updated the API to `AddPointAndTombstoneIterator` that adds a pair of point and range tombstone iters at a time. I did not consolidate `AddPointAndTombstoneIterator` and `AddIterator` yet to separate use cases that may or may not involve range tombstones. I'm thinking to consolidate them (or decide on not to consolidate them) as a next step after this PR.